modules/post/windows/gather/avast_memory_dump.rb

```diff
@@ -31,7 +31,19 @@ def initialize(info = {})
   end
 
   def avdump_exists?
-    file_exist?('C:\\Program Files\\Avast Software\\Avast\\AvDump.exe')
+    avdump_paths = [
+      'C:\\Program Files\\Avast Software\\Avast\\AvDump.exe',
+      'C:\\Program Files\\Avast Software\\BreachGuard\\AvDump.exe',
+      'C:\\Program Files\\Avast Software\\Cleanup\\AvDump.exe',
+      'C:\\Program Files\\Avast Software\\Driver Updater\\AvDump.exe',
+      'C:\\Program Files\\Avast Software\\SecureLine VPN\\AvDump.exe'
+    ]
+
+    avdump_paths.each do |p|
+      if file_exist?(p.to_s)
+        return p.to_s
+      end
+    end
```

Comment:
Whilst this is a nice idea, the idea of `avdump_exists?` is to check for a single file and return whether or not it exists on the system as a boolean value. This change effectively changes the `avdump_exists?` function to more of a `check_avdump_locations` function. Note that functions with a `?` mark at the end of them in Ruby are usually meant to return a boolean value by convention.

Reply:
Also note that as a result of your return statement, this will just return the first file that exists rather than enumerating all of the files, which is what I think you wanted to do? I could be wrong on your intention though, in which case feel free to correct me 👍 You would also need to update the statements below.

Reply:
> Note that functions with a `?` mark at the end of them in Ruby are usually meant to return a boolean value by convention.

I was unaware of this. It seems redundant to have two functions here with regards to checking if a file exists and then also checking again and returning a path for execution.

Would it be preferred to just rename the _avdump_exists?_ to something more explicit with regards to identify the file path and then returning the first value?

Reply:
Yep just wanted to have the name of the function reflect conventions. So since its no longer returning a boolean it shouldn't have a `?` in its name. You can rename it as you see fit so long as its descriptive and doesn't contain a `?` in its name.

Reply:
Totally makes sense. Thanks for the feedback! I'll update tonight.

Reply:
Thanks new name looks good :) Marking as resolved.